apps/fipsinstall.c

```diff
@@ -285,6 +296,15 @@ int fipsinstall_main(int argc, char **argv)
         case OPT_OUT:
             out_fname = opt_arg();
             break;
+        case OPT_SELF_TEST_NO_LOG:
+            self_test_log = 0;
+            break;
+        case OPT_SELF_TEST_CORRUPT_DESC:
```

Comment:
can you have both corruptions at the same time?

Reply:
The callback gets passed 3 fields currently
(1) phase (e,g: begin, corrupt, pass, fail)
(2) desc
(3) type

The user callback decides what to do with these fields.

desc could possibly be used by more that 1 self test, so it could corrupt multiple tests.
if both type and desc are set then it checks for both being true (i.e its an AND operation).

Reply:
Perhaps one way to integrate both callbacks is to pass a void* that is a structure containing the custom info?  I dunno, wait for Levitte to reply :)

Reply:
Changed to use OSSL_CALLBACK

Reply:
Callback is the same as other callback mechanisms using OSSL_CALLBACK